### PR TITLE
[BD-21] Make use of new-style waffle switch objects

### PR DESCRIPTION
**Pull request by regisb**
**TL;DR -** Stop the `from waffle_utils import WaffleSwitch` shenanigans.

**What changed?**

- WaffleSwitch is not imported from edx-platform's `waffle_utils` (which is deprecated), but from edx-toggles, which is thus added to the base requirements. Note that CourseWaffleFlag objects remain in edx-platform.
- Unit tests were refactored to take this into account.

This will be ready for review once edx-toggles==1.2.0 is released: https://github.com/edx/edx-toggles/pull/81 

**Developer Checklist**

**Before Merge**
- [x] Ensure edx-toggles 1.2.0 is released and tests are passing.
  - [x] Ensure 1.2.0 custom attributes show that no deprecated methods are still in use from WaffleSwitch.
- [x] Reviewed the [release process](./release_process.md)
~~- [ ] Translations up to date~~
~~- [ ] JS minified, SASS compiled~~
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

**Post Merge**
- [ ] Release new version.

JIRA: [JIRA-BD-21](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943/BD-21+Toggles+Settings+Documentation)

FIY: @edx/masters-devs-gta @robrap

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @regisb! I've created [OSPR-5113](https://openedx.atlassian.net/browse/OSPR-5113) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: openassessment/xblock/test/test_config_mixin.py [hunk 1]
```diff
@@ -139,32 +118,21 @@ def _run_feature_toggle_test(
...
-
         settings_feature_key = FEATURE_TOGGLES_BY_FLAG_NAME[flag_name]
-        with self.settings(FEATURES={settings_feature_key: settings_input}):
-            self.assertEqual(expected_output, getattr(my_block, feature_property, None))
+
+        with mock.patch('openassessment.xblock.config_mixin.WaffleSwitch', autospec=True) as MockWaffleSwitch:
+            MockWaffleSwitch.return_value.is_enabled.return_value = waffle_switch_input
+            with mock.patch(
+                'openassessment.xblock.config_mixin.import_course_waffle_flag', autospec=True
+            ) as mock_waffle_flag:
```
Comment (robrap) [inline openassessment/xblock/test/test_config_mixin.py:127]: Nit: `mock_course_waffle_flag` would be more consistent.

**Review by robrap — APPROVED** [review]
Thanks.  I'm going to add some pre-merge tasks to the PR description.

**Comment by regisb** [comment]
This will be ready to merge once [edx-toggles==1.2.0](https://pypi.org/project/edx-toggles/) is released.

File: openassessment/xblock/config_mixin.py [hunk 2]
```diff
@@ -21,16 +22,6 @@
 }
 
 
-def import_waffle_switch():
```
Comment (robrap) [inline openassessment/xblock/config_mixin.py:24]: Can we add a custom attribute in a separate PR/version to ensure this is not used by edx-platform before removing.  In other words, deprecate first, check, then remove.

**Review by robrap — COMMENTED** [review]
Note this needs a version bump.  I know you didn't necessarily say this was ready for review yet. :)

**Comment by regisb** [comment]
This should be ready to merge.
